Header header
frontier_node[] frontier_list